wgengine/router/dns/manager_windows.go

```diff
@@ -35,14 +36,76 @@ func newManager(mconfig ManagerConfig) managerImpl {
 	}
 }
 
-// keyOpenTimeout is how long we wait for a registry key to
-// appear. For some reason, registry keys tied to ephemeral interfaces
-// can take a long while to appear after interface creation, and we
-// can end up racing with that.
-const keyOpenTimeout = time.Minute
+const (
+	// keyOpenTimeout is how long we wait for a registry key to
+	// appear. For some reason, registry keys tied to ephemeral interfaces
+	// can take a long while to appear after interface creation, and we
+	// can end up racing with that.
+	keyOpenTimeout = time.Minute
+
+	// REG_NOTIFY_CHANGE_NAME notifies the caller if a subkey is added or deleted.
+	REG_NOTIFY_CHANGE_NAME uint32 = 0x00000001
+)
+
+func openKeyWait(k registry.Key, path string, access uint32, timeout time.Duration) (registry.Key, error) {
```

Comment:
We need to put any code copied from wireguard in its own file with the wireguard license at the top. (Perhaps later it'll get upstreamed into x/sys/windows.)

Reply:
OK, will do it in a minute.